kotlin/iwa-sample-app/src/main/java/com/esri/android/samples/iwasampleapp/MainActivity.kt

```diff
@@ -0,0 +1,173 @@
+                val hostname = dialogView.auth_hostname
+                val username = dialogView.auth_username
+                val password = dialogView.auth_password
+                // Set click listeners
+                builder.setPositiveButton("Sign In") {dialog, which ->
+                            // Create user credential
+                            credential = UserCredential(username.text.toString(), password.text.toString())
+                            signal.countDown()
+                        }
+                        .setNegativeButton("Cancel") {dialog, which ->
+                            // User cancel the loading process
+                            val remoteResource = challenge.remoteResource
+                            if (remoteResource is Loadable) {
+                                remoteResource.cancelLoad()
+                            }
+                            signal.countDown()
+                        }
+                        .setView(dialogView)
+                // Set message text
+                if (challenge.failureCount > 0) {
+                    hostname.setText("Wrong credential was passed to ${challenge.remoteResource.uri}")
+                } else {
+                    hostname.setText("Credential is required to access ${challenge.remoteResource.uri}")
+                }
+                
+                val dialog: AlertDialog = builder.create()
+                dialog.setCanceledOnTouchOutside(false)
+                dialog.show()
+                // Apply the button texts and disable the positive button unless both username and password contain text
+                dialog.getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = false
+                val watcher = object : TextWatcher {
+                    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
+                    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
+                    override fun afterTextChanged(s: Editable) {
+                        dialog.getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = 
+                                (username.text.length > 0) && (password.text.length > 0)
+                    }
+                }
+                username.addTextChangedListener(watcher)
+                password.addTextChangedListener(watcher)
+            })
+            signal.await()
+
+            // If credentials were set, return a new auth challenge response with them. otherwise, act like it was a cancel
+            if (credential != null) {
+                return AuthenticationChallengeResponse(
+                        AuthenticationChallengeResponse.Action.CONTINUE_WITH_CREDENTIAL, credential)
+            }
+        }
+
+        // No credentials were set , return a new auth challenge response with a cancel
+        return AuthenticationChallengeResponse(AuthenticationChallengeResponse.Action.CANCEL, challenge)
```

Comment:
I think this is inconsistent to what we show in the OAuth sample. Both samples should either return null or ` AuthenticationChallengeResponse(AuthenticationChallengeResponse.Action.CANCEL, challenge)`

Reply:
I have updated the OAuth sample to return ` AuthenticationChallengeResponse(AuthenticationChallengeResponse.Action.CANCEL, challenge)`